findcoordinator.go

```diff
@@ -2,8 +2,120 @@ package kafka
 
 import (
 	"bufio"
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/segmentio/kafka-go/protocol/findcoordinator"
+)
+
+// CoordinatorKeyType is used to specify the type of coordinator to look for
+type CoordinatorKeyType int8
+
+const (
+	// CoordinatorKeyTypeConsumer type is used when looking for a Group coordinator
+	CoordinatorKeyTypeConsumer CoordinatorKeyType = 0
+
+	// CoordinatorKeyTypeTransaction type is used when looking for a Transaction coordinator
+	CoordinatorKeyTypeTransaction CoordinatorKeyType = 1
 )
 
+// FindCoordinatorRequest is the request structure for the FindCoordinator function
+type FindCoordinatorRequest struct {
+	// Address of the kafka broker to send the request to.
+	Addr net.Addr
+
+	// The coordinator key.
+	Key string
+
+	// The coordinator key type. (Group, transaction, etc.)
+	KeyType CoordinatorKeyType
+}
+
+// FindCoordinatorResponseCoordinator contains details about the found coordinator
+type FindCoordinatorResponseCoordinator struct {
+	// NodeID holds the broker id.
+	NodeID int32
+
+	// Host of the broker
+	Host string
+
+	// Port on which broker accepts requests
+	Port int32
```

Comment:
We've taken the approach of not exposing implementation details like the size of the integer types in other public APIs, let's stay consistent?
```suggestion
	// NodeID holds the broker id.
	NodeID int

	// Host of the broker
	Host string

	// Port on which broker accepts requests
	Port int
```

Reply:
Roger that.